### PR TITLE
[misc] [bug] Fix legacy benchmarks/run.py

### PR DESCRIPTION
**Pull request by yolo2themoon**
Related issue = #

### CODE REVIEW

File: benchmarks/async_cases.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 import taichi as ti
 
-sys.path.append(os.path.join(ti.core.get_repo_dir(), 'tests', 'python'))
+sys.path.append(os.path.join(sys.path[0], '../', 'tests', 'python/'))
```
Comment (qiao-bo) [inline benchmarks/async_cases.py:7]: Can we briefly comment here on the path ;)? Before it was clear where it points to, now we depend on sys.path[0]?

**Review by qiao-bo — COMMENTED** [review]
Thanks!

File: benchmarks/async_cases.py (hunk 1 above)
Comment (yolo2themoon) [reply benchmarks/async_cases.py:7]: `ti.core.get_repo_dir()` now returns the directory`/home/username/.taichi/`,
and `sys.path[0]` points to the directory where the current file is located.

Comment (qiao-bo) [reply benchmarks/async_cases.py:7]: maybe `__file__` is better? `sys.path` is a list of search path i believe.

Comment (yolo2themoon) [reply benchmarks/async_cases.py:7]: Thanks for the tip, when trying to use `__file__`, I found that the absolute path was missing
sys.path[0] = `/home/username/.../taichi_dir/benchmarks`
__file__ = `run.py`

Comment (yolo2themoon) [reply benchmarks/async_cases.py:7]: How about we use `os.path.dirname(os.path.realpath(__file__))`, which also serves the purpose

**Review by qiao-bo — APPROVED** [review]
Thanks!

**Pull request merged**